### PR TITLE
PPTP-1003 : Support for partnership type selection

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**
We still only support general partnerships for the time being. Organisation type selection also updated.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/PartnershipDetails.scala [hunk 1]
```diff
@@ -16,34 +16,15 @@
...
 
 case class PartnershipDetails(
-  sautr: String,
-  postcode: String,
-  override val registration: IncorporationRegistrationDetails
-) extends RegistrationDetails with GrsEntityDetails
+  partnershipType: PartnershipTypeEnum,
+  generalPartnershipDetails: Option[GeneralPartnershipDetails] = None
+  // TODO: other partnership type details here
+)
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/PartnershipDetails.scala:26]: I'm trying to follow the same pattern here as that already adopted for business types: ie. a type and a bunch of Optional type details fields.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/GeneralPartnershipConnector.scala [hunk 2]
```diff
@@ -21,19 +21,22 @@ import play.api.http.Status.CREATED
...
 }
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class PartnershipConnector @Inject() (httpClient: HttpClient, config: AppConfig, metrics: Metrics)(
-  implicit ec: ExecutionContext
-) extends GrsIntegration[PartnershipCreateJourneyRequest, PartnershipDetails] {
+class GeneralPartnershipConnector @Inject() (
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/GeneralPartnershipConnector.scala:34]: @kiranprakashdigital, you might want to consider refactoring this Connector to support all partnership types. Although, when I attempted to support all GRS integrations (in `GrsIntegration`) I failed due to issues with types and json de/serialisation.

**Comment by PauloPortugal** [comment]
Also commit d3b76bc1660c8e184c57c0fbfa3a6d10c4e2af7e and d3b76bc1660c8e184c57c0fbfa3a6d10c4e2af7e have the same subject, so potentially a `git commit --amend` at the time (where you amend the previous commit with new code that still fit the intent of the commit).

Now you would have to rebase it interactively.
Not a blocker.

**Comment by RyanKellyHMRC** [comment]
To be honest, I don't feel very confident using rebase when pushes have already been made. Perhaps you could find some time to give us (assuming that @kiranprakashdigital feels the same way) a session on its use and the potential pitfalls?

**Review by PauloPortugal — APPROVED** [review]

**Pull request merged**